unified-codes/apps/web/src/actions/DetailActions.ts

```diff
@@ -3,21 +3,58 @@ import { Action } from 'redux';
 
 export const DETAIL_ACTIONS = {
   UPDATE_ENTITY: 'detail/updateEntity',
+  FETCH_ENTITY: 'detail/fetchEntity',
+  FETCH_ENTITY_SUCCESS: 'detail/fetchEntitySuccess',
+  FETCH_ENTITY_FAILURE: 'detail/fetchEntityFailure',
 };
 
 export interface IDetailUpdateEntityAction extends Action<string> {
   entity: IEntity;
 }
 
-export type IDetailAction = IDetailUpdateEntityAction;
+export interface IDetailFetchEntityAction extends Action<string> {
+  code: string;
+}
+
+export interface IDetailFetchSuccessAction extends Action<string> {
+  entity: IEntity;
+}
+
+export interface IDetailFetchFailureAction extends Action<string> {
+  error: Error;
+}
```

Comment:
I think the `FetchEntityAction` is redundant (as every time the entity is updated a fetch should probably occur?). Can probably go without and just rename the success and failures as `IDetailUpdateSuccessAction` etc.? If we need a separate action can always add it when needed 🤷 .

Reply:
Yeah I wasn't 100% about having a separate action or not. I think you're right though - since we don't currently have a reason to have them separate.

Reply:
Updated in d5c829867